extensions/markdown-language-features/src/test/smartSelect.test.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+import 'mocha';
+
+import MarkdownSmartSelect from '../features/smartSelect';
+import { InMemoryDocument } from './inMemoryDocument';
+import { createNewMarkdownEngine } from './engine';
+
+const testFileName = vscode.Uri.file('test.md');
+
+suite.only('markdown.SmartSelect', () => {
+	test('Smart select single word', async () => {
+		const selections = await getSelectionRangesForDocument(`Hello`, new vscode.Position(0, 0));
```

Comment:
Instead of having to hardcode the position you are testing, it might be fun to come up a way to say in the string where the cursor is. Take look at https://github.com/microsoft/vscode/blob/87b6d03182f3ad2a0f4688a077b26e467719b906/extensions/typescript-language-features/src/test/jsDocCompletions.test.ts#L53 for an example of this

This would potentially make the tests easier to read:

```ts
await getSelectionRangesForDocument(`${CURSOR}Hello`);
```